spring-core/src/main/java/org/springframework/core/annotation/AnnotationsScanner.java

```diff
@@ -323,9 +321,7 @@ private AnnotationsScanner() {
 		if (superclass != Object.class && superclass != null) {
 			R superclassResult = processMethodHierarchy(context, aggregateIndex,
 					superclass, processor, classFilter, rootMethod, includeInterfaces);
-			if (superclassResult != null) {
-				return superclassResult;
-			}
+			return superclassResult;
```

Comment:
Although this may be technically equivalent, it breaks with the structure of the rest of the method.

As such, I do not think the change is warranted.

So, please revert this change.